index.js

```diff
@@ -16,6 +16,139 @@ cookieButton.addEventListener('click', () => {
      cookie.classList.add('hidden');
 })
 
+//active navbar elements
+
+function activateMenuLink(allLinks, linkToActivate) { //funkcja dodajaca active do linkow  (pierwszy parametr lista wszystkich linkow, w drugim link ktory chce aktywowac)
+     for(let i = 0; i < allLinks.length; i++){ //petla dla calej listy linkow
+
+          let link = allLinks[i]; // element w tablicy all links
+
+          if (link.classList.contains('active')) { //  jesli jest active to usuwamy
+               link.classList.remove("active");
+          }
+     }
+
+     if (!linkToActivate.classList.contains('active')) { // jesli nie zawiera active to dodajemy
+          linkToActivate.classList.add("active");
+     }
+
+     let href = linkToActivate.getAttribute("href"); //pobieramy atrybut href (np #features) 
+
+     window.history.pushState(href, "Vitamina - "+href, href); // dodajemy go to historii przegladarki- zmienia sie url bez reloadu
+}
+
+const navigationMenuContainer = document.querySelector(".navigation");
+const navigationMenuElement = document.querySelector(".navigation__menu__list");
+const navigationMenuElementFocus = document.querySelectorAll(".navigation__menu__list__link");
+
+for (let i = 0; i < navigationMenuElementFocus.length; i++) {
+          navigationMenuElementFocus[i].addEventListener("click", (event) => {
+          activateMenuLink(navigationMenuElementFocus, event.target); //wywolujemy funkcje na kliku
+     });
+}
+
+// scroll handler
+
+let linksDestination = document.getElementsByClassName("linkLoc"); // zbieramy elementy majace klase linkLoc w HTML
+let links = document.getElementsByClassName("navigation__menu__list__link"); 
+let positionsList = []; // tworzymy pusta liste 
+let linkHrefs = []; // tworzymy pusta liste hfefow
+
+for(let i = 0; i < links.length; i++){
+     let link = links[i]; // sprawdzamy kazdy element
```

Comment:
Zobacz, uzywasz zmiennej link tylko w jednym miejscu, czy jest potrzeba robic zmiennej do tego? Jesli tak, to mogłaby być chociaż const :)